packages/network/src/settled-tracking-middleware.ts

```diff
@@ -15,7 +15,31 @@ export interface FetchDebugInfo {
   url: string;
 }
 
-const TrackingInfoPerFetch: WeakMap<Fetch, FetchDebugInfo[]> = new WeakMap();
+const TrackingInfoPerFetchSymbol = Symbol.for(
+  '_DATA_EDEN_TRACKING_INFO_PER_FETCH'
+);
+
+const TrackingInfoPerFetch: WeakMap<Fetch, FetchDebugInfo[]> = (() => {
+  // in some bundling contexts (e.g. webpack as used by ember-auto-import) you might
+  // end up with multiple copies of this module in the final bundle. In that case
+  // we need to ensure that we have a single global tracking map. We do this by
+  // using a symbol that is the same across all copies, and then using that symbol
+  // to store the map on the global object.
+  //
+  // this ensures that at any given time (regardless of how many copies of this module exist)
+  // we can always get the same map (and therefore our API's work properly).
+
+  // TODO: figure out how to make this work without the casting `globalThis`
+  let _global = globalThis as Record<string | symbol, unknown>;
+  let trackingInfoPerFetch = _global[TrackingInfoPerFetchSymbol];
```

Comment:
@rwjblue sent me here; [here's a playground][1] with how I would write this. A cast is necessary because TS seems to special-case `globalThis`, but you can write a cast which is totally safe *and* totally localized.

[1]: https://www.typescriptlang.org/play?#code/C4TwDgpgBAYhwGMAWUC8VSQPYDMo-mQG4AoEiADzCwCdgoBLAO2AhpwEMFo5EkARCACMArgHMAkkxxYoAbxJQoAZ2BcA1gH4AXCuA1mY0ktUc6AFQYBbCLqYirQtsag3gSLABNdqg0yOKSiI0ADY++oakAL5kCFhMqlDmNBqGUjIACmy8yADKII5YIWhQ+YUhAHQyNAAUgQDkAPr8AILmLY0AovydAHKN5gBKLQDCANISvQDijZMwAPKNGZ2DjTCd5iMAEvUkAJSksfGJyan+6VhZNDlIugDqEBzqALIcYAA8NwA0sIQCwuILgBtAC6AD4SjUans0BCFEoAPQI0otdbmACauncDGUUAQHESOKgHBCAHcOCBcSEIGIGMBrBxWFAnPiRMpoKToGZoMgIAh1IZ8LQMEhoGAaBB2UxuIEkVBcMSmPKwPT4iSoOKsJA6CAKlAAEIgKBswUcKAhLD4kIMABeEE8eIJ9A4TAdmC5TCYWDUqqYP05xJCyiwsuRVie0GxuPFDAAbozoJ54BwGMUxBB7MwICEjTH46xdHSoJ4sJKoF76NSnhhZFbqTRcTV3ByKaHicW+SEzIyGPFXGx01BSVgRCFPHsKoFqfRGmILUJ1eg51gFyFzEgiQSMOAIArl6v10SAGTyQJKIGnflpaSXbJ-MpCIognRQB5PV4fb6-PiCUSSG+gmCLhRIcSjThgKRXucN5XDcJSzvOJIXpBArQZkd58A+T4uAweA1AAhPoZz-uh1x-DC8JKBBxEXLBfzwfuSGXqhJG3mRmEFI+IQgiUTAQKSr6PC8bzQsBZBKBKwDBEqRFQaxdF8NEeyiSQQA